### PR TITLE
Add/show validation notice for unvalidated recovery emails

### PR DESCRIPTION
**Pull request by southp**
## Summary
This PR adds the unvalidated warning notice right below the recovery email if it is not unvalidated.
image, along with a "resend" button.

Before this PR, we don't have a way to validate the ownership / correctness of recovery emails entered by users, causing them to lose a reliable option for account recovery. 

## Test plan
Before testing, please make sure you have deployed D3185.

1. Log in as a user with a unvalidated recovery email set.
    1. Visit http://calypso.localhost:3000/me/security/checkup , and see the unvalidated warning notice.
    1. Click "resend", and check your mailbox to find the validation email
    1. Click the "Confirm" button, and your recovery email should be validated successfully.
    1. Refresh the security checkup page, and see the unvalidated warning gone.
1. Log in as a user without a recovery email set
    1. The unvalidated warning notice shouldn't be there.
    1. Add a recovery email.
    1. Check your mailbox, you should receive a validation email before hitting the resend button.
1. Log in as a user with a validated recovery email set.
    1. The unvalidated warning notice shouldn't be there.
    1. Update the recovery email.
    1. Check your mailbox, you should receive a validation email.

### CODE REVIEW

File: client/me/security-checkup/recovery-email-validation-notice.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+class RecoveryEmailValidationNotice extends Component {
```
Comment (jordwest) [inline client/me/security-checkup/recovery-email-validation-notice.jsx]: Is there or will there be a `RecoveryPhoneValidationNotice` components? Could they be combined into one with a `type` prop or similar?

Comment (southp) [inline client/me/security-checkup/recovery-email-validation-notice.jsx]: Yes, you read my mind: https://github.com/Automattic/wp-calypso/blob/582713f831cde99141e10e52a36be6466adfa3ea/client/me/security-checkup/recovery-phone-validation-notice.jsx

These two are quite different. The phone one contains a form for inputing validation code, and thus contains extra components and needs a few more action creators. At the moment, I couldn't see an obvious way to combine the two. How do you think? If you got any idea, I'd love to try :)

**Comment by southp** [comment]
> Now when I enter a recovery SMS, the message comes through to my phone with a validation code, but its not clear where I should enter that code?

Oops. That was my mistake. The patch D3185 contains the code to send validation code to the SMS number you entered, but the flow is implemented in #10226. This PR is only about the email part :)

Once this is deployed, I will make sure only the validation email part of the backend code is deployed. So our users won't have this confusion.

**Comment by jordwest** [comment]
Ok cool, looks good to 🚢  then!

**Pull request merged**